[BUG] EmbeddingModel.embedAll() fails when input has more than 128 items

<!-- Please provide as many details as possible, this will help us to deliver a fix as soon as possible. Thank you! -->

**Describe the bug**
When calling `embedAll()` with more than 128 TextSegments, the request fails with a ValidationException from AWS Bedrock Cohere API. The API enforces a maximum of 128 items in the texts field, but embedAll() does not automatically handle batching internally.

**Log and Stack trace**
```shell
Caused by: software.amazon.awssdk.services.bedrockruntime.model.ValidationException: Malformed input request: #/texts: expected maximum item count: 128, found: 200, please reformat your input and try again. (Service: BedrockRuntime, Status Code: 400, Request ID: c89e481c-aae4-4b73-97e0-14d6fa2758a4) (SDK Attempt Count: 1)
	at software.amazon.awssdk.services.bedrockruntime.model.ValidationException$BuilderImpl.build(ValidationException.java:152)
	at software.amazon.awssdk.services.bedrockruntime.model.ValidationException$BuilderImpl.build(ValidationException.java:100)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.utils.RetryableStageHelper.retryPolicyDisallowedRetryException(RetryableStageHelper.java:168)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.RetryableStage.execute(RetryableStage.java:73)
	at software.amazon.awssdk.core.internal.http.pipeline.stages.RetryableStage.execute(RetryableStage.java:36)
	at software.amazon.awssdk.core.internal.http.pipeline.RequestPipelineBuilder$ComposingRequestPipelineStage.execute(RequestPipelineBuilder.java:206)
	at software.amazon.awssdk.core.internal.http.StreamManagingStage.execute(StreamManagingStage.java:53)
	at software.amazon.awssdk.core.internal.http.StreamManagingStage.execute(StreamManagingStage.java:35)
```

**To Reproduce**
```java
package io.github.esgai;

import dev.langchain4j.data.embedding.Embedding;
import dev.langchain4j.data.segment.TextSegment;
import dev.langchain4j.model.bedrock.BedrockCohereEmbeddingModel;
import dev.langchain4j.model.output.Response;
import software.amazon.awssdk.regions.Region;

import java.util.List;
import java.util.stream.IntStream;

public class EsgAiApplication {

    public static void main(String[] args) {
        BedrockCohereEmbeddingModel embeddingModel = BedrockCohereEmbeddingModel
                .builder()
                .region(Region.US_EAST_1)
                .model("cohere.embed-multilingual-v3")
                .inputType(BedrockCohereEmbeddingModel.InputType.SEARCH_QUERY)
                .build();

        var data = IntStream.range(0, 200)
                .mapToObj(i -> TextSegment.textSegment("test" + i))
                .toList();

        Response<List<Embedding>> embedded = embeddingModel.embedAll(data);
        System.out.println(embedded);
    }
}

```

**Expected behavior**
* embedAll() should either:
   -   1. Automatically batch requests into groups of 128 (or the maximum allowed by the provider), and then return a combined list of embeddings.

   - 2. Or throw a clear exception early in client-side code, instead of sending an invalid request to the API.

**Please complete the following information:**
- LangChain4j version: 1.4.0
- LLM(s) used: AWS Bedrock
- Java version: 21

**Additional context**
<!-- Add any other context about the problem here. -->

--- a/langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/BedrockCohereEmbeddingModel.java
+++ b/langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/BedrockCohereEmbeddingModel.java
@@ -1,22 +1,5 @@
 package dev.langchain4j.model.bedrock;
 
-import dev.langchain4j.data.embedding.Embedding;
-import dev.langchain4j.data.segment.TextSegment;
-import dev.langchain4j.model.embedding.DimensionAwareEmbeddingModel;
-import dev.langchain4j.model.output.Response;
-import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
-import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
-import software.amazon.awssdk.core.SdkBytes;
-import software.amazon.awssdk.regions.Region;
-import software.amazon.awssdk.services.bedrockruntime.BedrockRuntimeClient;
-import software.amazon.awssdk.services.bedrockruntime.model.InvokeModelRequest;
-import software.amazon.awssdk.services.bedrockruntime.model.InvokeModelResponse;
-
-import java.nio.charset.Charset;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-
 import static dev.langchain4j.internal.RetryUtils.withRetryMappingExceptions;
 import static dev.langchain4j.internal.Utils.getOrDefault;
 import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
@@ -25,6 +8,23 @@ import static dev.langchain4j.model.bedrock.Json.toJson;
 import static java.util.Arrays.stream;
 import static java.util.stream.Collectors.toList;
 import static software.amazon.awssdk.regions.Region.US_EAST_1;
+
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.embedding.DimensionAwareEmbeddingModel;
+import dev.langchain4j.model.output.Response;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.bedrockruntime.BedrockRuntimeClient;
+import software.amazon.awssdk.services.bedrockruntime.model.InvokeModelRequest;
+import software.amazon.awssdk.services.bedrockruntime.model.InvokeModelResponse;
 
 /**
  * Bedrock Cohere embedding model with support for both versions:
@@ -35,11 +35,14 @@ import static software.amazon.awssdk.regions.Region.US_EAST_1;
  */
 public class BedrockCohereEmbeddingModel extends DimensionAwareEmbeddingModel {
 
+    private static final int DEFAULT_MAX_SEGMENTS_PER_BATCH = 96;
+
     private final BedrockRuntimeClient client;
     private final String model;
     private final String inputType;
     private final String truncate;
     private final int maxRetries;
+    private final int maxSegmentsPerBatch;
 
     public BedrockCohereEmbeddingModel(Builder builder) {
         this.client = getOrDefault(builder.client, () -> initClient(builder));
@@ -47,31 +50,38 @@ public class BedrockCohereEmbeddingModel extends DimensionAwareEmbeddingModel {
         this.inputType = ensureNotBlank(builder.inputType, "inputType");
         this.truncate = builder.truncate;
         this.maxRetries = getOrDefault(builder.maxRetries, 2);
+        this.maxSegmentsPerBatch = getOrDefault(builder.maxSegmentsPerBatch, DEFAULT_MAX_SEGMENTS_PER_BATCH);
     }
 
     private BedrockRuntimeClient initClient(Builder builder) {
         return BedrockRuntimeClient.builder()
                 .region(getOrDefault(builder.region, US_EAST_1))
-                .credentialsProvider(getOrDefault(builder.credentialsProvider,
-                        () -> DefaultCredentialsProvider.builder().build()))
+                .credentialsProvider(
+                        getOrDefault(builder.credentialsProvider, () -> DefaultCredentialsProvider.builder()
+                                .build()))
                 .build();
     }
 
     @Override
     public Response<List<Embedding>> embedAll(List<TextSegment> textSegments) {
+        List<Embedding> embeddings = new ArrayList<>();
 
-        Map<String, Object> requestParameters = toRequestParameters(textSegments);
-        String requestJson = toJson(requestParameters);
+        for (int i = 0; i < textSegments.size(); i += maxSegmentsPerBatch) {
+            List<TextSegment> batch = textSegments.subList(i, Math.min(textSegments.size(), i + maxSegmentsPerBatch));
+            Map<String, Object> requestParameters = toRequestParameters(batch);
+            String requestJson = toJson(requestParameters);
 
-        InvokeModelResponse invokeModelResponse = withRetryMappingExceptions(() ->
-                invoke(requestJson), maxRetries, BedrockExceptionMapper.INSTANCE);
+            InvokeModelResponse invokeModelResponse =
+                    withRetryMappingExceptions(() -> invoke(requestJson), maxRetries, BedrockExceptionMapper.INSTANCE);
 
-        String responseJson = invokeModelResponse.body().asUtf8String();
-        BedrockCohereEmbeddingResponse embeddingResponse = fromJson(responseJson, BedrockCohereEmbeddingResponse.class);
+            String responseJson = invokeModelResponse.body().asUtf8String();
+            BedrockCohereEmbeddingResponse embeddingResponse =
+                    fromJson(responseJson, BedrockCohereEmbeddingResponse.class);
 
-        List<Embedding> embeddings = stream(embeddingResponse.getEmbeddings().getFloatEmbeddings())
-                .map(Embedding::from)
-                .collect(toList());
+            embeddings.addAll(stream(embeddingResponse.getEmbeddings().getFloatEmbeddings())
+                    .map(Embedding::from)
+                    .toList());
+        }
 
         return Response.from(embeddings);
     }
@@ -106,6 +116,7 @@ public class BedrockCohereEmbeddingModel extends DimensionAwareEmbeddingModel {
         private Region region;
         private AwsCredentialsProvider credentialsProvider;
         private Integer maxRetries;
+        private Integer maxSegmentsPerBatch;
 
         public Builder model(Model model) {
             return model(model.getValue());
@@ -154,13 +165,17 @@ public class BedrockCohereEmbeddingModel extends DimensionAwareEmbeddingModel {
             return this;
         }
 
+        public Builder maxSegmentsPerBatch(Integer maxSegmentsPerBatch) {
+            this.maxSegmentsPerBatch = maxSegmentsPerBatch;
+            return this;
+        }
+
         public BedrockCohereEmbeddingModel build() {
             return new BedrockCohereEmbeddingModel(this);
         }
     }
 
     public enum Model {
-
         COHERE_EMBED_ENGLISH_V3("cohere.embed-english-v3"),
         COHERE_EMBED_MULTILINGUAL_V3("cohere.embed-multilingual-v3");
 
@@ -176,7 +191,6 @@ public class BedrockCohereEmbeddingModel extends DimensionAwareEmbeddingModel {
     }
 
     public enum InputType {
-
         SEARCH_DOCUMENT("search_document"),
         SEARCH_QUERY("search_query"),
         CLASSIFICATION("classification"),
@@ -194,7 +208,6 @@ public class BedrockCohereEmbeddingModel extends DimensionAwareEmbeddingModel {
     }
 
     public enum Truncate {
-
         NONE("NONE"),
         START("START"),
         END("END");

--- a/langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/BedrockCohereEmbeddingModelIT.java
+++ b/langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/BedrockCohereEmbeddingModelIT.java
@@ -1,17 +1,17 @@
 package dev.langchain4j.model.bedrock;
 
-import static dev.langchain4j.model.bedrock.common.BedrockAiServicesIT.sleepIfNeeded;
 import static dev.langchain4j.model.bedrock.BedrockCohereEmbeddingModel.InputType.SEARCH_QUERY;
 import static dev.langchain4j.model.bedrock.BedrockCohereEmbeddingModel.Model.COHERE_EMBED_ENGLISH_V3;
 import static dev.langchain4j.model.bedrock.BedrockCohereEmbeddingModel.Model.COHERE_EMBED_MULTILINGUAL_V3;
 import static dev.langchain4j.model.bedrock.BedrockCohereEmbeddingModel.Truncate.END;
+import static dev.langchain4j.model.bedrock.common.BedrockAiServicesIT.sleepIfNeeded;
 import static org.assertj.core.api.Assertions.assertThat;
 
 import dev.langchain4j.data.embedding.Embedding;
 import dev.langchain4j.data.segment.TextSegment;
 import dev.langchain4j.model.output.Response;
+import java.util.ArrayList;
 import java.util.List;
-
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
@@ -67,6 +67,75 @@ class BedrockCohereEmbeddingModelIT {
         assertThat(response.tokenUsage()).isNull();
         assertThat(response.finishReason()).isNull();
 
+        assertThat(embeddingModel.dimension()).isEqualTo(1024);
+    }
+
+    @Test
+    void should_use_default_size_for_large_batch() {
+        BedrockCohereEmbeddingModel embeddingModel = BedrockCohereEmbeddingModel.builder()
+                .model(COHERE_EMBED_ENGLISH_V3)
+                .inputType(SEARCH_QUERY)
+                .truncate(END)
+                .build();
+
+        assertThat(embeddingModel).isNotNull();
+
+        int numberOfSegments = 165;
+        List<TextSegment> segments = new ArrayList<>();
+        for (int i = 0; i < numberOfSegments; i++) {
+            segments.add(TextSegment.from("This is dummy text segment " + (i + 1)));
+        }
+
+        Response<List<Embedding>> response = embeddingModel.embedAll(segments);
+        assertThat(response).isNotNull();
+
+        List<Embedding> embeddings = response.content();
+
+        // Verify all segments were processed
+        assertThat(embeddings).hasSize(numberOfSegments);
+
+        // Verify each embedding has the correct dimension
+        for (Embedding embedding : embeddings) {
+            assertThat(embedding.vector()).hasSize(1024);
+        }
+
+        // Verify embeddings are not null or empty
+        assertThat(embeddings).allMatch(embedding -> embedding.vector() != null && embedding.vector().length > 0);
+
+        assertThat(response.tokenUsage()).isNull();
+        assertThat(response.finishReason()).isNull();
+        assertThat(embeddingModel.dimension()).isEqualTo(1024);
+    }
+
+    @Test
+    void should_use_custom_batch_size_for_large_batch() {
+        BedrockCohereEmbeddingModel embeddingModel = BedrockCohereEmbeddingModel.builder()
+                .model(COHERE_EMBED_ENGLISH_V3)
+                .inputType(SEARCH_QUERY)
+                .truncate(END)
+                .maxSegmentsPerBatch(45)
+                .build();
+
+        assertThat(embeddingModel).isNotNull();
+
+        int numberOfSegments = 102;
+        List<TextSegment> segments = new ArrayList<>();
+        for (int i = 0; i < numberOfSegments; i++) {
+            segments.add(TextSegment.from("This is dummy text segment " + (i + 1)));
+        }
+
+        Response<List<Embedding>> response = embeddingModel.embedAll(segments);
+        assertThat(response).isNotNull();
+
+        List<Embedding> embeddings = response.content();
+
+        assertThat(embeddings).hasSize(numberOfSegments);
+
+        // Verify embeddings exists for all textSegments
+        assertThat(embeddings).allMatch(embedding -> embedding.vector() != null && embedding.vector().length > 0);
+
+        assertThat(response.tokenUsage()).isNull();
+        assertThat(response.finishReason()).isNull();
         assertThat(embeddingModel.dimension()).isEqualTo(1024);
     }
 

--- a/langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/common/BedrockAiServicesIT.java
+++ b/langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/common/BedrockAiServicesIT.java
@@ -1,26 +1,16 @@
 package dev.langchain4j.model.bedrock.common;
 
-import static dev.langchain4j.model.bedrock.BedrockCohereEmbeddingModel.InputType.SEARCH_QUERY;
-import static dev.langchain4j.model.bedrock.BedrockCohereEmbeddingModel.Model.COHERE_EMBED_ENGLISH_V3;
-import static dev.langchain4j.model.bedrock.BedrockCohereEmbeddingModel.Truncate.END;
 import static dev.langchain4j.model.bedrock.TestedModels.AWS_NOVA_LITE;
 import static dev.langchain4j.model.bedrock.TestedModels.AWS_NOVA_MICRO;
 import static dev.langchain4j.model.bedrock.TestedModels.AWS_NOVA_PRO;
 import static dev.langchain4j.model.bedrock.TestedModels.CLAUDE_3_HAIKU;
 import static dev.langchain4j.model.bedrock.TestedModels.COHERE_COMMAND_R_PLUS;
 import static dev.langchain4j.model.bedrock.TestedModels.MISTRAL_LARGE;
-import static org.assertj.core.api.Assertions.assertThat;
 
-import dev.langchain4j.data.embedding.Embedding;
-import dev.langchain4j.data.segment.TextSegment;
-import dev.langchain4j.model.bedrock.BedrockCohereEmbeddingModel;
 import dev.langchain4j.model.chat.ChatModel;
-import dev.langchain4j.model.output.Response;
 import dev.langchain4j.service.common.AbstractAiServiceIT;
-import java.util.ArrayList;
 import java.util.List;
 import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
 
 @EnabledIfEnvironmentVariable(named = "AWS_SECRET_ACCESS_KEY", matches = ".+")
@@ -35,75 +25,6 @@ public class BedrockAiServicesIT extends AbstractAiServiceIT {
     @AfterEach
     void afterEach() {
         sleepIfNeeded();
-    }
-
-    @Test
-    void should_use_default_size_for_large_batch() {
-        BedrockCohereEmbeddingModel embeddingModel = BedrockCohereEmbeddingModel.builder()
-                .model(COHERE_EMBED_ENGLISH_V3)
-                .inputType(SEARCH_QUERY)
-                .truncate(END)
-                .build();
-
-        assertThat(embeddingModel).isNotNull();
-
-        int numberOfSegments = 165;
-        List<TextSegment> segments = new ArrayList<>();
-        for (int i = 0; i < numberOfSegments; i++) {
-            segments.add(TextSegment.from("This is dummy text segment " + (i + 1)));
-        }
-
-        Response<List<Embedding>> response = embeddingModel.embedAll(segments);
-        assertThat(response).isNotNull();
-
-        List<Embedding> embeddings = response.content();
-
-        // Verify all segments were processed
-        assertThat(embeddings).hasSize(numberOfSegments);
-
-        // Verify each embedding has the correct dimension
-        for (Embedding embedding : embeddings) {
-            assertThat(embedding.vector()).hasSize(1024);
-        }
-
-        // Verify embeddings are not null or empty
-        assertThat(embeddings).allMatch(embedding -> embedding.vector() != null && embedding.vector().length > 0);
-
-        assertThat(response.tokenUsage()).isNull();
-        assertThat(response.finishReason()).isNull();
-        assertThat(embeddingModel.dimension()).isEqualTo(1024);
-    }
-
-    @Test
-    void should_use_custom_batch_size_for_large_batch() {
-        BedrockCohereEmbeddingModel embeddingModel = BedrockCohereEmbeddingModel.builder()
-                .model(COHERE_EMBED_ENGLISH_V3)
-                .inputType(SEARCH_QUERY)
-                .truncate(END)
-                .maxSegmentsPerBatch(45)
-                .build();
-
-        assertThat(embeddingModel).isNotNull();
-
-        int numberOfSegments = 102;
-        List<TextSegment> segments = new ArrayList<>();
-        for (int i = 0; i < numberOfSegments; i++) {
-            segments.add(TextSegment.from("This is dummy text segment " + (i + 1)));
-        }
-
-        Response<List<Embedding>> response = embeddingModel.embedAll(segments);
-        assertThat(response).isNotNull();
-
-        List<Embedding> embeddings = response.content();
-
-        assertThat(embeddings).hasSize(numberOfSegments);
-
-        // Verify embeddings exists for all textSegments
-        assertThat(embeddings).allMatch(embedding -> embedding.vector() != null && embedding.vector().length > 0);
-
-        assertThat(response.tokenUsage()).isNull();
-        assertThat(response.finishReason()).isNull();
-        assertThat(embeddingModel.dimension()).isEqualTo(1024);
     }
 
     public static void sleepIfNeeded() {

--- a/langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/common/BedrockAiServicesIT.java
+++ b/langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/common/BedrockAiServicesIT.java
@@ -1,18 +1,27 @@
 package dev.langchain4j.model.bedrock.common;
 
-import dev.langchain4j.model.chat.ChatModel;
-import dev.langchain4j.service.common.AbstractAiServiceIT;
-import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
-
-import java.util.List;
-
+import static dev.langchain4j.model.bedrock.BedrockCohereEmbeddingModel.InputType.SEARCH_QUERY;
+import static dev.langchain4j.model.bedrock.BedrockCohereEmbeddingModel.Model.COHERE_EMBED_ENGLISH_V3;
+import static dev.langchain4j.model.bedrock.BedrockCohereEmbeddingModel.Truncate.END;
 import static dev.langchain4j.model.bedrock.TestedModels.AWS_NOVA_LITE;
 import static dev.langchain4j.model.bedrock.TestedModels.AWS_NOVA_MICRO;
 import static dev.langchain4j.model.bedrock.TestedModels.AWS_NOVA_PRO;
 import static dev.langchain4j.model.bedrock.TestedModels.CLAUDE_3_HAIKU;
 import static dev.langchain4j.model.bedrock.TestedModels.COHERE_COMMAND_R_PLUS;
 import static dev.langchain4j.model.bedrock.TestedModels.MISTRAL_LARGE;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.bedrock.BedrockCohereEmbeddingModel;
+import dev.langchain4j.model.chat.ChatModel;
+import dev.langchain4j.model.output.Response;
+import dev.langchain4j.service.common.AbstractAiServiceIT;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
 
 @EnabledIfEnvironmentVariable(named = "AWS_SECRET_ACCESS_KEY", matches = ".+")
 public class BedrockAiServicesIT extends AbstractAiServiceIT {
@@ -26,6 +35,75 @@ public class BedrockAiServicesIT extends AbstractAiServiceIT {
     @AfterEach
     void afterEach() {
         sleepIfNeeded();
+    }
+
+    @Test
+    void should_use_default_size_for_large_batch() {
+        BedrockCohereEmbeddingModel embeddingModel = BedrockCohereEmbeddingModel.builder()
+                .model(COHERE_EMBED_ENGLISH_V3)
+                .inputType(SEARCH_QUERY)
+                .truncate(END)
+                .build();
+
+        assertThat(embeddingModel).isNotNull();
+
+        int numberOfSegments = 165;
+        List<TextSegment> segments = new ArrayList<>();
+        for (int i = 0; i < numberOfSegments; i++) {
+            segments.add(TextSegment.from("This is dummy text segment " + (i + 1)));
+        }
+
+        Response<List<Embedding>> response = embeddingModel.embedAll(segments);
+        assertThat(response).isNotNull();
+
+        List<Embedding> embeddings = response.content();
+
+        // Verify all segments were processed
+        assertThat(embeddings).hasSize(numberOfSegments);
+
+        // Verify each embedding has the correct dimension
+        for (Embedding embedding : embeddings) {
+            assertThat(embedding.vector()).hasSize(1024);
+        }
+
+        // Verify embeddings are not null or empty
+        assertThat(embeddings).allMatch(embedding -> embedding.vector() != null && embedding.vector().length > 0);
+
+        assertThat(response.tokenUsage()).isNull();
+        assertThat(response.finishReason()).isNull();
+        assertThat(embeddingModel.dimension()).isEqualTo(1024);
+    }
+
+    @Test
+    void should_use_custom_batch_size_for_large_batch() {
+        BedrockCohereEmbeddingModel embeddingModel = BedrockCohereEmbeddingModel.builder()
+                .model(COHERE_EMBED_ENGLISH_V3)
+                .inputType(SEARCH_QUERY)
+                .truncate(END)
+                .maxSegmentsPerBatch(45)
+                .build();
+
+        assertThat(embeddingModel).isNotNull();
+
+        int numberOfSegments = 102;
+        List<TextSegment> segments = new ArrayList<>();
+        for (int i = 0; i < numberOfSegments; i++) {
+            segments.add(TextSegment.from("This is dummy text segment " + (i + 1)));
+        }
+
+        Response<List<Embedding>> response = embeddingModel.embedAll(segments);
+        assertThat(response).isNotNull();
+
+        List<Embedding> embeddings = response.content();
+
+        assertThat(embeddings).hasSize(numberOfSegments);
+
+        // Verify embeddings exists for all textSegments
+        assertThat(embeddings).allMatch(embedding -> embedding.vector() != null && embedding.vector().length > 0);
+
+        assertThat(response.tokenUsage()).isNull();
+        assertThat(response.finishReason()).isNull();
+        assertThat(embeddingModel.dimension()).isEqualTo(1024);
     }
 
     public static void sleepIfNeeded() {

